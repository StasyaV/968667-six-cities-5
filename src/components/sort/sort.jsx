import React from "react";
import PropTypes from "prop-types";
import {ActionCreator} from '../../store/action';
import {connect} from 'react-redux';
import {SortType} from "../../const";

const Sort = (props) => {
  const {currentSort, updateSort, openSort} = props;

  const onSortClick = (evt) => {
    evt.preventDefault();
    updateSort(evt.target.textContent);
  };

  return (
    <ul onClick={onSortClick} className={`places__options places__options--custom ${openSort ? `places__options--opened` : ``}`}>
      <li className={`places__option ${SortType.POPULAR === currentSort ? `places__option--active` : ``}`} tabIndex="0">{SortType.POPULAR}</li>
      <li className={`places__option ${SortType.PRICE_LOW_TO_HIGH === currentSort ? `places__option--active` : ``}`} tabIndex="0">{SortType.PRICE_LOW_TO_HIGH}</li>
      <li className={`places__option ${SortType.PRICE_HIGH_TO_LOW === currentSort ? `places__option--active` : ``}`} tabIndex="0">{SortType.PRICE_HIGH_TO_LOW}</li>
      <li className={`places__option ${SortType.RATING === currentSort ? `places__option--active` : ``}`} tabIndex="0">{SortType.RATING}</li>
    </ul>
  );

};

Sort.propTypes = {
  updateSort: PropTypes.func.isRequired,
  currentSort: PropTypes.string.isRequired,
  openSort: PropTypes.bool.isRequired,
};

const mapStateToProps = (({currentSort, openSort}) => ({
  currentSort,
  openSort
}));

const mapDispatchToProps = ((dispatch) => ({
  updateSort(sort) {
    dispatch(ActionCreator.updateSort(sort));
  },
}));

export {Sort};
export default connect(mapStateToProps, mapDispatchToProps)(Sort);
