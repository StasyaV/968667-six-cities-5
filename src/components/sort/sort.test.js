import React from "react";
import renderer from "react-test-renderer";
import {Sort} from "./sort";

const noop = () => {};

describe(`Sort render`, () => {
  it(`Should Sort render correctly`, () => {
    const tree = renderer
    .create(<Sort
      currentSort={`Popular`}
      openSort={false}
      updateSortAction={noop}
    />,
    {
      createNodeMock: () => {
        return {};
      }
    }
    )
    .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
