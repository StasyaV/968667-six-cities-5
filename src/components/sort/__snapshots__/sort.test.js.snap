// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sort render Should Sort render correctly 1`] = `
<ul
  className="places__options places__options--custom "
  onClick={[Function]}
>
  <li
    className="places__option places__option--active"
    tabIndex="0"
  >
    Popular
  </li>
  <li
    className="places__option "
    tabIndex="0"
  >
    Price: low to high
  </li>
  <li
    className="places__option "
    tabIndex="0"
  >
    Price: high to low
  </li>
  <li
    className="places__option "
    tabIndex="0"
  >
    Top rated first
  </li>
</ul>
`;
